### PR TITLE
Direct round trips

### PR DESCRIPTION
**Pull request by gabriel-indik**
Signed-off-by: Gabriel Indik <gabriel.indik@kaleido.io>

### CODE REVIEW

**Comment by gabriel-indik** [comment]
Changes in this PR:
If the sender and recipient ID match (round trip) no mTLS communication is established. Instead DX will:
1. In the case of messages, directly place the message in the queue and emit the corresponding events
2. In the case of blob transfers, copy the blob to the "received" folder location (which may contain the destination) and emit the corresponding events.

File: src/routers/api.ts [hunk 1]
```diff
@@ -152,46 +154,38 @@ router.post('/messages', async (req, res, next) => {
...
+      if (senderID !== peerID) {
+        throw new RequestError(`Sender ID mismatch expected=${peerID} recieved=${senderDestination}`, 400);
       }
-      if (segments.length > 1) {
-        if (!config.destinations?.includes(segments[1])) {
-          throw new RequestError(`Unknown sender destination expected=${config.destinations?.join('|') ?? 'none'} recieved=${segments[1]}`, 400);
-        }
-        senderDestination = segments[1];
+      if (senderDestination !== undefined && !config.destinations?.includes(senderDestination)) {
+        throw new RequestError(`Unknown sender destination expected=${config.destinations?.join('|') ?? 'none'} recieved=${senderDestination}`, 400);
```
Comment (awrichar) [inline src/routers/api.ts:163]: typo: `recieved` -> `received`

**Review by awrichar — APPROVED** [review]
Looks great. I verified that this seems to fix the "self signed certificate" error seen in the tests on https://github.com/hyperledger/firefly/pull/942.

**Review by peterbroadhurst — APPROVED** [review]
👍

**Pull request merged**